components/datadog/apps/nginx/k8s/modifiers.go

```diff
@@ -0,0 +1,164 @@
+package k8s
+
+import (
+	appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/apps/v1"
+	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/core/v1"
+	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/meta/v1"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+// DeploymentModifier is a function that operates on a DeploymentArgs struct
+type DeploymentModifier func(args *appsv1.DeploymentArgs)
+
+// rawMapToStringMap converts map[string]string
+func rawMapToStringMap(raw map[string]string) pulumi.StringMap {
+	result := pulumi.StringMap{}
+	for k, v := range raw {
+		result[k] = pulumi.String(v)
+	}
+	return result
+}
+
+// mergeStringMaps merges two pulumi.StringMaps into one
+func mergeStringMaps(left, right pulumi.StringMap) pulumi.StringMap {
+	// buffer
+	merged := pulumi.StringMap{}
+	// get everything from left map
+	for k, v := range left {
+		merged[k] = v
+	}
+	// get everything from right map, this will overwrite if
+	// keys are duplicated
+	for k, v := range right {
+		merged[k] = v
+	}
```

Comment:
The [`maps` package](https://pkg.go.dev/maps) of the recent Go versions allows something more concise.

```suggestion
	// get everything from left map
	merged := maps.Clone(left)
	// get everything from right map, this will overwrite if
	// keys are duplicated
	maps.Copy(merged, right)
```